app/revenue/management/commands/kudos_revenue.py

```diff
@@ -0,0 +1,139 @@
+                logger.error(f"* Error during API call => {rs['message']}")
+            return
+        print(f"Total {params['action']} found in blocks {params['startblock']} to {params['endblock']}: {len(records)}")
+        for rec in records:
+            save_incoming_tx(network, rec)
+    except Exception as e:
+        logger.error(f"* Exception in kudos_revenue => {e}")
+
+
+def save_incoming_tx(network, record):
+    decimals = 18. if len(record['contractAddress']) == 0 else float(record['tokenDecimal'])
+    dgp = DigitalGoodPurchase.objects.create(
+        emails=[],
+        tokenName='ETH' if len(record['contractAddress']) == 0 else record['tokenSymbol'],
+        tokenAddress=record['contractAddress'],
+        amount=(int(record['value']) / decimals) if decimals > 0 else int(record['value']),
+        ip='',
+        network=network,
+        from_address=record['from'],
+        receive_address=record['to'],
+        metadata=record,
+        txid=record['hash'],
+        receive_txid=record['hash'],
+    )
+    dgp.update_tx_status()
+    dgp.update_receive_tx_status()
+    dgp.save()
+
+
+class Command(BaseCommand):
+
+    help = 'tracks kudos purchases'
+
+    def add_arguments(self, parser):
+        parser.add_argument('network', default='mainnet', type=str)
+        parser.add_argument(
+            '--account-address',
+            default=REVENUE_ACCOUNT_ADDRESS,
+            type=str,
+            help="Account address to track revenue"
+        )
+        parser.add_argument(
+            '--start-block',
+            default=DEFAULT_START_BLOCK,
+            type=int,
+            help="Start at block number"
+        )
+        parser.add_argument(
+            '--end-block',
+            default=99999999999,
+            type=int,
+            help="End at block number"
+        )
+
+    def handle(self, *args, **options):
+        if settings.ETHERSCAN_API_KEY == '':
+            logger.error("* Environment variable ETHERSCAN_API_KEY has not been set")
+            return
+        # config
+        network = options['network']
+        params  = {
+            'module': 'account',
+            'address': options['account_address'],
+            'startblock': options['start_block'],
+            'endblock': options['end_block'],
+            'apikey': settings.ETHERSCAN_API_KEY,
+            'sort': 'asc'
+        }
+
+        # Get incoming ETH transactions
+        dgp_records = DigitalGoodPurchase.objects \
+            .filter(receive_address__iexact=options['account_address'], tokenName__iexact='ETH') \
+            .order_by('-id') \
+            .values('id', 'metadata')[:1]
+        
+        if len(dgp_records) > 0:
+            params['startblock'] = int(dgp_records[0]['metadata']['blockNumber']) + 1
+            logger.info(f"ETH Tx Records were found. Continuing from block {params['startblock']}")
+        
+        params['action'] = 'txlist'
+        call_etherscan_api(network, params)
```

Comment:
why do we need to call `call_etherscan_api()` twice?  why not just once

Reply:
Etherscan API uses two separate methods to retrieve ETH transactions (txlist) and ERC-20 token transactions (tokentx)